src/Features/Core/Portable/DesignerAttribute/DesignerAttributeDiscoveryService.cs

```diff
@@ -104,13 +101,44 @@ private static async ValueTask<bool> HasDesignerCategoryTypeAsync(Project projec
             }
         }
 
+        public async ValueTask ProcessPriorityDocumentAsync(
```

Comment:
to make things clearer, i broke this into two entrypoints the host can call into.  One to analyze a specific document, so that when a user is editing  document, we quickly sync changes with it to the UI.  Another to analyze the whole solution.